n3fit/src/evolven3fit_new/evolve.py

```diff
@@ -0,0 +1,221 @@
+import pathlib
+import logging
+import sys
+
+import numpy as np
+from reportengine.compat import yaml
+
+from ekobox import genpdf, gen_info
+from ekomark import apply
+from eko import basis_rotation as br
+from eko import output
+from validphys.api import API
+
+from . import utils, eko_utils
+
+
+log = logging.getLogger(__name__)
+
+LOG_FILE = "evolven3fit_new.log"
+
+LOGGING_SETTINGS = {"formatter" : "%(asctime)s %(name)s/%(levelname)s: %(message)s",
+                    "level" : logging.INFO
+}
+
+def evolve_fit(
+    conf_folder,
+    q_fin,
+    q_points,
+    op_card_dict,
+    t_card_dict,
+    force,
+    eko_path=None,
+    dump_eko=None,
+):
+    """
+    Evolves all the fitted replica in conf_folder/nnfit
+
+    Parameters
+    ----------
+
+        conf_folder: str or pathlib.Path
+            path to the folder containing the fit
+        q_fin: float
+            final point of the q_grid
+        q_points: int
+            number of points in the q_grid
+        op_card_dict: dict
+            user settings for the op_card
+        t_card_dict: dict
+            user settings for the t_card
+        force: bool
+            whether to force the evolution to be done again
+        eko_path: str or pathlib.Path
+            path where the eko is stored (if None the eko will be
+            recomputed)
+        dump_eko: str or pathlib.Path
+            path where the eko is dumped (if None the eko won't be
+            stored)
+    """
+    log_file = pathlib.Path(conf_folder) / LOG_FILE
+    if log_file.exists():
+        if force:
+            log_file.unlink()
+        else:
+            raise FileExistsError(
+                f"Log file already exists: {log_file} evolven3fit_new has already been run?"
+            )
+    log_file = logging.FileHandler(log_file)
+    stdout_log = logging.StreamHandler(sys.stdout)
+    for log in [log_file, stdout_log]:
+        log.setLevel(LOGGING_SETTINGS["level"])
+        log.setFormatter(
+            logging.Formatter(LOGGING_SETTINGS["formatter"])
+        )
+    for logger in (log, *[logging.getLogger("eko")]):
```

Comment:
Using `log` here, you will receive the last element of the previous iteration, but I believe you want the module level object.
```suggestion
    for logger in (_logger, *[logging.getLogger("eko")]):
```